File: frontend/src/api/skill.ts
import { get, post, put, del } from './request'
import type {
  Skill,
  SkillTemplate,
  SkillTestResult,
  SkillCategory,
  SkillFunction
} from '@/types'

/**
 * 获取 Skill 列表
 */
export function getSkillList(params?: {
  category?: SkillCategory
  function?: SkillFunction
  enabled?: boolean
}): Promise<Skill[]> {
  return get<Skill[]>('/skills', params)
}

/**
 * 获取 Skill 模板列表
 */
export function getSkillTemplates(): Promise<SkillTemplate[]> {
  return get<SkillTemplate[]>('/skills/templates')
}

/**
 * 获取 Skill 详情
 */
export function getSkillDetail(id: number): Promise<Skill> {
  return get<Skill>(`/skills/${id}`)
}

/**
 * 创建 Skill
 */
export function createSkill(
  data: Partial<Skill>
): Promise<Skill> {
  return post<Skill>('/skills', data)
}

/**
 * 从模板创建 Skill
 */
export function createSkillFromTemplate(templateId: string): Promise<Skill> {
  return post<Skill>('/skills/from-template', { template_id: templateId })
}

/**
 * 更新 Skill
 */
export function updateSkill(id: number, data: Partial<Skill>): Promise<Skill> {
  return put<Skill>(`/skills/${id}`, data)
}

/**
 * 删除 Skill
 */
export function deleteSkill(id: number): Promise<void> {
  return del<void>(`/skills/${id}`)
}

/**
 * 启用/禁用 Skill
 */
export function toggleSkill(id: number): Promise<Skill> {
  return put<Skill>(`/skills/${id}/toggle`)
}

/**
 * 测试 Skill 路由
 */
export function testSkill(id: number, question: string): Promise<SkillTestResult> {
  return post<SkillTestResult>(`/skills/${id}/test`, { question })
}

File: backend/app/routers/skill.py
"""Skill 管理路由"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.skill import SkillCreate, SkillUpdate, SkillTestRequest
from app.services import skill_service, skill_router
from app.utils.response import success, APIError
from app.utils.auth import require_auth

router = APIRouter(dependencies=[Depends(require_auth)])


@router.get("")
def list_skills(
    category: str = None,
    function: str = None,
    enabled: str = None,
    db: Session = Depends(get_db),
):
    """Skill 列表"""
    enabled_val = None
    if enabled is not None and enabled != "":
        enabled_val = enabled.lower() == "true"
    items = skill_service.get_list(db, category=category, function=function, enabled=enabled_val)
    return success(items)


@router.get("/templates")
def get_templates():
    """模板列表"""
    return success(skill_service.get_templates())


@router.get("/{sid}")
def get_skill(sid: int, db: Session = Depends(get_db)):
    """Skill 详情"""
    s = skill_service.get_by_id(db, sid)
    return success(skill_service._to_out(s))


@router.post("")
def create_skill(data: SkillCreate, db: Session = Depends(get_db)):
    """创建 Skill"""
    s = skill_service.create(db, data)
    return success(skill_service._to_out(s), message="创建成功")


@router.post("/from-template")
def create_from_template(template_id: str = "", db: Session = Depends(get_db)):
    """从模板创建"""
    if not template_id:
        raise APIError("请提供 template_id")
    s = skill_service.create_from_template(db, template_id)
    return success(skill_service._to_out(s), message="从模板创建成功")


@router.put("/{sid}")
def update_skill(sid: int, data: SkillUpdate, db: Session = Depends(get_db)):
    """更新 Skill"""
    s = skill_service.update(db, sid, data)
    return success(skill_service._to_out(s), message="更新成功")


@router.delete("/{sid}")
def delete_skill(sid: int, db: Session = Depends(get_db)):
    """删除 Skill"""
    skill_service.delete(db, sid)
    return success(message="删除成功")


@router.put("/{sid}/toggle")
def toggle_skill(sid: int, db: Session = Depends(get_db)):
    """启用/禁用"""
    s = skill_service.toggle(db, sid)
    return success(skill_service._to_out(s), message="切换成功")


@router.post("/{sid}/test")
async def test_skill(sid: int, data: SkillTestRequest, db: Session = Depends(get_db)):
    """测试 Skill 路由"""
    # 确认 Skill 存在
    skill_service.get_by_id(db, sid)
    result = await skill_router.test_route(data.question, db)
    return success({
        "matched_skill": result.matched_skill,
        "match_type": result.match_type,
        "score": result.score,
        "all_scores": result.all_scores,
    })
